src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/staves/FireStaff.java

```diff
@@ -0,0 +1,84 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.magical.staves;
+
+import io.github.bakedlibs.dough.protection.Interaction;
+import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
+import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import io.github.thebusybiscuit.slimefun4.implementation.items.LimitedUseItem;
+import org.bukkit.*;
+import org.bukkit.entity.Player;
+import org.bukkit.event.entity.FoodLevelChangeEvent;
+import org.bukkit.inventory.ItemStack;
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+public class FireStaff extends LimitedUseItem {
+
+    public static final int MAX_USES = 120;
+    private static final NamespacedKey usageKey = new NamespacedKey(Slimefun.instance(), "firestaff_usage");
+
+    @ParametersAreNonnullByDefault
+    public FireStaff(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(itemGroup, item, recipeType, recipe);
+
+        setMaxUseCount(MAX_USES);
+    }
+
+    @Override
+    protected @Nonnull NamespacedKey getStorageKey() {
+        return usageKey;
+    }
+
+    @Nonnull
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            Player p = e.getPlayer();
+            ItemStack item = e.getItem();
+
+            if (p.getFoodLevel() >= 4 || p.getGameMode() == GameMode.CREATIVE) {
+                // Get a target block with max. 7 blocks of distance
+                Location loc = p.getTargetBlock(null, 7).getLocation();
+
+                if (loc.getWorld() != null && loc.getChunk().isLoaded()) {
+                    if (loc.getWorld().getPVP() && Slimefun.getProtectionManager().hasPermission(p, loc, Interaction.PLACE_BLOCK)) {
+                        e.cancel();
+                        useItem(p, item, loc);
+                    } else {
+                        Slimefun.getLocalization().sendMessage(p, "messages.cannot-place", true);
+                    }
+                }
+            } else {
+                Slimefun.getLocalization().sendMessage(p, "messages.hungry", true);
+            }
+        };
+    }
+
+    @ParametersAreNonnullByDefault
+    private void useItem(Player p, ItemStack item, Location loc) {
+        World world = loc.getWorld();
+        if (world != null) {
+            p.getTargetBlock(null, 7).setType(Material.FIRE);
```

Comment:
Does this not create issues? Like making wood instantly disappear and spawning the flame in its place, being able to spawn fire flying in the air (which I don't know if it would even crash the server), and I guess this is more style points than anything but it would be cool to have soul fire instead of normal fire on soulsand and sulsoil

Reply:
Also may be quicker to do loc.getBlock() instead, since you already pass that in, instead of getting TargetBlock. Not sure how impactful that would be though

Reply:
i could try

Reply:
now you mentioned, they could also break bedrock with this right?

Reply:
Might want to do an air check and check if under the air or on the side since flames can be on sides. Is a block and for the side a block that can catch fire.

Reply:
can you check discord panda?